src/engine/email.go

```diff
@@ -0,0 +1,118 @@
+package engine
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+
+	"github.com/scoville/scvl/src/domain"
+)
+
+// CreateEmailRequest is the request for creating an email
+type CreateEmailRequest struct {
+	IsPreview      bool
+	SpreadsheetURL string
+	SheetName      string
+	Sender         string
+	Template       string
+	Title          string
+	User           *domain.User
+}
+
+// SendEmailRequest is the request for sending an email
+type SendEmailRequest struct {
+	ToAddresses  []*string
+	BccAddresses []*string
+	Body         string
+}
+
+// CreateEmail creates an email
+func (e *Engine) CreateEmail(req *CreateEmailRequest) (emailTemplate *domain.EmailTemplate, err error) {
+
+	emailTemplate, err = createEmailTemplate(e, req)
+	return
+}
+
+// SendEmail send a email
+func (e *Engine) SendEmail(req *CreateEmailRequest) (err error) {
+	emailTemplate, err := createEmailTemplate(e, req)
```

Comment:
You should call this after you update `req.Template`. Otherwise, updating it has no effects